File: frontend/src/app/modules/userdashboard/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http : HttpClient) { }

  public getquizzes() {
    return this.http.get('http://localhost:8080/quiz/get-all');
  }

  public getactivequizzes(){
    return this.http.get('http://localhost:8080/quiz/get-all/active')
  }

  public getquiz(id: any){
    return this.http.get(`http://localhost:8080/quiz/get/${id}`);
  }

  public getquizbycategory(id: any){
    return this.http.get(`http://localhost:8080/quiz/get/category/${id}`);
  }

  public getactivequizbycategory(id: any){
    return this.http.get(`http://localhost:8080/quiz/get/category/active/${id}`);
  }

  public addquiz(data: any) {
    return this.http.post('http://localhost:8080/quiz/add', data);
  }

  public deleteQuiz(id: any){
    return this.http.delete(`http://localhost:8080/quiz/delete/${id}`)
  }

  public updateQuiz(quiz: any){
    return this.http.put('http://localhost:8080/quiz/update', quiz);
  }
}

File: backend/src/main/java/com/exam/backend/services/QuizService.java
package com.exam.backend.services;

import com.exam.backend.entity.Category;
import com.exam.backend.entity.Quiz;

import java.util.Set;

public interface QuizService {

    public Quiz addQuiz(Quiz quiz);

    public Quiz updateQuiz(Quiz quiz);

    public Set<Quiz> getAllQuiz();

    public Quiz getQuiz(Long id);

    public Set<Quiz> getQuizzesByCategory(Category category);

    public Set<Quiz> getActiveQuizzes(Boolean active);

    public Set<Quiz> getActivedQuizzesByCategory(Category category, Boolean active);

    public void deleteQuiz(Long id);
}
